app/assets/javascripts/routers/data_portal.js

```diff
@@ -28,6 +29,37 @@
         iso: iso,
         year: +year
       });
+    },
+
+    report: function (params) {
+      // Don't forget to stop the router on each route
+      // otherwise you'll break the browser's back button
+      Backbone.history.stop();
+      var indicators = [],
+        serializedIndicators;
+
+      if (params) {
+        var indicatorsb64 = params.split('=')[1] ? params.split('=')[1] : null;
+        if (indicatorsb64) {
+          // decodes b64 string to object
+          indicators = JSON.parse(App.Helper.URL.decode(indicatorsb64).indicators);
+
+          // deserializes incoming indicators
+          serializedIndicators = indicators.map(function(ind) {
```

Comment:
If the widgets are deserialised, then the variable can't be named `serializedIndicators`.